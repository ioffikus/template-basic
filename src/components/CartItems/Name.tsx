import React from 'react';
import { Link } from 'react-router-dom';

interface IProps {
  uid: string;
  shortDescription: string;
  name: string;
}

interface IState {}

class Copyright extends React.Component<IProps, IState> {
  render() {
    const { uid, shortDescription, name } = this.props;
    return (
      <div className="root">
        <style jsx>{`
          .root {
            font-size: 16px;
            line-height: 1.3;
            font-weight: bold;
            overflow: hidden;
            text-overflow: ellipsis;
            white-space: nowrap;
          }

          .root :global(.name) {
            color: inherit;
            white-space: nowrap;
          }
          .root :global(.code) {
            color: #b2bdcb;
            font-size: 12px;
          }
        `}</style>
        <Link to={`/product/${uid}`} className="name" title={shortDescription}>
          {name}
        </Link>
        <div className="code"># {uid}</div>
      </div>
    );
  }
}

export default Copyright;
