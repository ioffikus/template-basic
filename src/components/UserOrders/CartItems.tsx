import React from 'react';
import { Link } from 'react-router-dom';
import { Table } from 'antd';
import _ from 'lodash';
import Price from 'src/components/Price';
import CartProductImage from 'src/components/CartProductImage';

interface IRecord {
  key: number;
  pictureUrl: string;
  name: string;
  count: number;
  price: number;
  currency: string;
  uid: string;
}

interface IProps {
  ui: Alicanto.Models.UI;
  items: Alicanto.Models.CatalogItem[];
}

interface IState {}

class CartItems extends React.Component<IProps, IState> {
  getDataSource = (): IRecord[] => {
    return this.props.items.map(item => {
      const key = item.id;
      const uid = item.uid;
      const schemaFields = item.schema_fields;
      const pictureUrl = _.get(schemaFields, 'picture_url.value', '');
      const name = _.get(schemaFields, 'name.value', '');
      const count = _.get(item, 'count', 0);
      const currency = _.get(schemaFields, 'currency.value', '');
      const price = _.get(schemaFields, 'price.value', 0);
      return { key, pictureUrl, name, count, price, currency, uid };
    });
  };

  render() {
    const commonUI = this.props.ui.common;
    const columns = [
      {
        title: commonUI.tables.cells.image.i18n.defaultText,
        key: 'image',
        render: (text: string, record: IRecord) => {
          return <CartProductImage pictureUrl={record.pictureUrl} imageAlt={record.name} />;
        },
      },
      {
        title: commonUI.tables.cells.name.i18n.defaultText,
        key: 'name',
        render: (text: string, record: IRecord) => {
          return <Link to={`/product/${record.uid}`}>{record.name}</Link>;
        },
      },
      { title: commonUI.tables.cells.count.i18n.defaultText, dataIndex: 'count' },
      {
        title: commonUI.tables.cells.price.i18n.defaultText,
        dataIndex: 'price',
        render: (text: string, record: IRecord) => {
          const { price, currency } = record;
          return <Price price={price} currency={currency} />;
        },
      },
    ];

    return (
      <div className="root">
        <style jsx>{`
          .root :global(.ant-table-row:last-child td) {
            border: 0;
          }
        `}</style>
        <Table columns={columns} dataSource={this.getDataSource()} pagination={false} />
      </div>
    );
  }
}

export default CartItems;
