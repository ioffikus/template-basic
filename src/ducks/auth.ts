import { createAction, createReducer } from 'redux-act';
import { AxiosResponse } from 'axios';
import { push } from 'connected-react-router';
import { GET, POST } from 'src/core/API';
import EndpointsConfig from 'src/configs/endpoints';

export const REDUCER = 'AUTH';
const NS = `${REDUCER}__`;

export interface IAuthState extends Store.IReduxCommonState {
  isAuthenticated: boolean;
  isAuthLinkRequesting: boolean;
  url: string;
  data: object;
}

export const initialState: IAuthState = {
  isAuthenticated: false,
  isRequesting: false,
  isRequestError: false,
  isAuthLinkRequesting: false,
  url: null,
  data: null,
};

const reducer = createReducer<IAuthState>({}, initialState);

export const loginUserSuccess = createAction(`${NS}LOGIN_USER_SUCCESS`);
reducer.on(loginUserSuccess, state => ({
  ...state,
  isRequesting: false,
  isRequestError: false,
  isAuthenticated: true,
}));

const authLinkSuccess = createAction<{ url: string; data: object }>(`${NS}AUTH_LINK_SUCCESS`);
reducer.on(authLinkSuccess, (state, { url, data }) => ({
  ...state,
  url,
  data: { ...data },
  isAuthLinkRequesting: false,
}));

const loginUserFailure = createAction(`${NS}LOGIN_USER_FAILURE`);
reducer.on(loginUserFailure, state => ({
  ...state,
  isRequesting: false,
  isAuthenticated: false,
  isRequestError: true,
}));

const authLinkFailure = createAction(`${NS}AUTH_LINK_FAILURE`);
reducer.on(authLinkFailure, state => ({
  ...state,
  isAuthLinkRequesting: false,
  url: null,
  data: null,
}));

const loginUserRequest = createAction(`${NS}LOGIN_USER_REQUEST`);
reducer.on(loginUserRequest, state => ({
  ...state,
  isRequesting: true,
}));

const authLinkRequest = createAction(`${NS}AUTH_LINK_REQUEST`);
reducer.on(authLinkRequest, state => ({
  ...state,
  isAuthLinkRequesting: true,
  url: null,
  data: null,
}));

const reset = createAction(`${NS}RESET`);
reducer.on(reset, () => ({
  ...initialState,
}));

export const getAuthLink = (): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    dispatch(authLinkRequest());
    GET(
      {
        url: `${EndpointsConfig.AUTH.AUTH_LINK}`,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        const { url, data } = resp.data as { url: string; data: object };
        // WIP does not use Alicanto.API.GET_AUTH_OAUTH_PROVIDER_URL_RESPONSE for develop
        dispatch(authLinkSuccess({ url, data }));
        resolve();
      })
      .catch(error => {
        dispatch(authLinkFailure());
        reject(error);
      });
  });
};

export const loginUser = (username: string, password: string): Store.IReduxThunkCb => (
  dispatch,
  getState,
  xShopHeaders,
) => {
  return new Promise((resolve, reject) => {
    dispatch(loginUserRequest());
    const data: Alicanto.API.POST_AUTH_LOGIN_BODY = {
      username,
      password,
    };
    POST(
      {
        data,
        url: `${EndpointsConfig.AUTH.LOGIN}`,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        dispatch(loginUserSuccess());
        if (process.env.BROWSER) {
          // TODO: move to external
          // redirect to home page
          dispatch(push('/'));
        }
        resolve();
      })
      .catch(error => {
        dispatch(loginUserFailure());
        reject(error);
      });
  });
};

export const logoutUser = (cb: () => void): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    dispatch(loginUserRequest());
    GET(
      {
        url: `${EndpointsConfig.AUTH.LOGOUT}`,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        dispatch(loginUserSuccess());
        dispatch(reset());
        if (cb) {
          cb();
        }
        if (process.env.BROWSER) {
          // TODO: move to external
          // redirect to home page
          dispatch(push('/'));
        }
        resolve();
      })
      .catch(error => {
        dispatch(loginUserFailure());
        reject(error);
      });
  });
};

export default reducer;
