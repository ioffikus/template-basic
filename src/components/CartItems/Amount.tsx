import React from 'react';
import { connect } from 'react-redux';
import { InputNumber, Spin } from 'antd';
import dispatch from 'src/store/dispatch';
import { increaseQuantity, decreaseQuantity } from 'src/ducks/cart';

interface IOwnProps {
  mayChangeAmountItems: boolean;
  uid: string;
  count: number;
  pendingShowSpinning?: number;
  pendingHideSpinning?: number;
}

interface IConnectedState {}

type IProps = IOwnProps & IConnectedState;

interface IState {
  isSpinning: boolean;
}

const mapStateToProps = (state: Store.IState) => ({});

const WrappedAmount = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class Amount extends React.Component<IProps, IState> {
    static defaultProps = {
      pendingShowSpinning: 400,
      pendingHideSpinning: 300,
    };

    spinningTimeoutId: number;

    state: IState = {
      isSpinning: false,
    };

    showSpinning = () => {
      if (!!this.spinningTimeoutId) {
        window.clearTimeout(this.spinningTimeoutId);
        this.spinningTimeoutId = null;
      }
      this.spinningTimeoutId = window.setTimeout(() => {
        this.setState({ isSpinning: true });
      }, this.props.pendingShowSpinning);
    };

    hideSpinning = () => {
      if (!!this.spinningTimeoutId) {
        window.clearTimeout(this.spinningTimeoutId);
        this.spinningTimeoutId = null;
      }
      if (this.state.isSpinning) {
        this.spinningTimeoutId = window.setTimeout(() => {
          this.setState({ isSpinning: false });
        }, this.props.pendingHideSpinning);
      }
    };

    handleChangeAmount = (value: number) => {
      if (process.env.BROWSER) {
        const { uid, count } = this.props;
        this.showSpinning();
        if (count < value) {
          dispatch(
            increaseQuantity(uid, () => {
              this.hideSpinning();
            }),
          );
        } else {
          dispatch(
            decreaseQuantity(uid, () => {
              this.hideSpinning();
            }),
          );
        }
      }
    };

    componentWillUnmount() {
      if (process.env.BROWSER) {
        if (!!this.spinningTimeoutId) {
          window.clearTimeout(this.spinningTimeoutId);
        }
      }
    }

    render() {
      const { mayChangeAmountItems, count } = this.props;
      return (
        <div className="root">
          <style jsx>{`
            .root {
              display: inline-block;
              vertical-align: top;
            }
            .root :global(.ant-input-number) {
              border: 0;
              display: flex;
              justify-content: center;
              background: transparent;
              color: #062f49;
            }
            .root :global(.ant-input-number:hover) {
              border-color: transparent;
            }
            .root :global(.ant-input-number-focused) {
              box-shadow: none;
              border-color: transparent;
            }
            .root :global(.ant-input-number-handler-wrap) {
              border: 0;
              opacity: 1;
              width: 100%;
              display: flex;
              flex-direction: row-reverse;
              background: transparent;
            }
            .root :global(.ant-input-number-input-wrap) {
              position: relative;
              z-index: 3;
              width: 32px;
              height: 32px;
            }
            .root :global(.ant-input-number-handler) {
              width: 50%;
              height: 100%;
              border: 0;
              top: 0;
              bottom: 0;
              background: transparent;
            }
            .root :global(.ant-input-number-handler:hover) {
              height: 100% !important;
            }
            .root :global(.ant-input-number-handler-down-inner) {
              right: auto;
              left: 4px;
            }
            .root :global(.ant-input-number-input) {
              background-color: #e5ecf4;
              border-radius: 8px;
              color: inherit;
              text-align: right;
              font-weight: bold;
            }
          `}</style>
          {mayChangeAmountItems ? (
            <Spin spinning={this.state.isSpinning}>
              <InputNumber min={1} value={count} onChange={this.handleChangeAmount} />
            </Spin>
          ) : (
            count
          )}
        </div>
      );
    }
  },
);

export default WrappedAmount;
