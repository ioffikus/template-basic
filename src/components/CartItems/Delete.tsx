import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'antd';
import dispatch from 'src/store/dispatch';
import { deleteFromCart } from 'src/ducks/cart';
import Svg from 'src/components/Svg';

interface IOwnProps {
  mayDeleteItems: boolean;
  uid: string;
}

interface IConnectedState {}

type IProps = IOwnProps & IConnectedState;

interface IState {}

const mapStateToProps = (state: Store.IState) => ({});

const WrappedDelete = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class Delete extends React.Component<IProps, IState> {
    handleDelete = (uid: string) => () => {
      dispatch(deleteFromCart(uid));
    };

    render() {
      const { mayDeleteItems, uid } = this.props;
      return (
        <div className="root">
          <style jsx>{`
            .root {
              text-align: right;
            }
            .root :global(.button-delete) {
              background: transparent;
              color: #b2bdcb;
            }
            .root :global(.icon-svg-trash) {
              height: 15px;
              width: 14px;
            }
          `}</style>
          {mayDeleteItems ? (
            <Button shape="circle" onClick={this.handleDelete(uid)} className="button-delete">
              <Svg name="trash" className="icon-svg-trash" />
            </Button>
          ) : (
            <span />
          )}
        </div>
      );
    }
  },
);

export default WrappedDelete;
