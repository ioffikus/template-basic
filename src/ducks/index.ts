// instead this, we have router in store via connected-react-router
// import { routerReducer as routing, RouterState } from 'react-router-redux';
import { combineReducers } from 'redux';
import auth from './auth';
import cart from './cart';
import cartPaidItems from './cartPaidItems';
import catalogQuery from './catalogQuery';
import catalog from './catalog';
import categories from './categories';
import categoriesMenu from './categoriesMenu';
import filter from './filter';
import locale from './locale';
import meta from './meta';
import page from './page';
import product from './product';
import progress from './progress';
import popularItems from './popularItems';
import recommendItems from './recommendItems';
import routes from './routes';
import ui from './ui';
import user from './user';
import userDelivery from './userDelivery';
import userOrders from './userOrders';

export default combineReducers<Store.IState>({
  ui,
  auth,
  user,
  categories,
  categoriesMenu,
  catalogQuery,
  catalog,
  product,
  progress,
  routes,
  meta,
  filter,
  cart,
  cartPaidItems,
  userDelivery,
  userOrders,
  popularItems,
  recommendItems,
  page,
  locale,
});
