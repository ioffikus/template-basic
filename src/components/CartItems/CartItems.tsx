import React from 'react';
import { connect } from 'react-redux';
import { Table } from 'antd';
import cx from 'classnames';
import _ from 'lodash';
import Price from 'src/components/Price';
import CartProductImage from 'src/components/CartProductImage';
import Name from './Name';
import Amount from './Amount';
import Delete from './Delete';

interface IRecord {
  pictureUrl: string;
  nameFields: {
    name: string;
    shortDescription: string;
  };
  uid: string;
  key: string;
  count: number;
  cost: {
    currency: string;
    price: number;
  };
}

interface IOwnProps {
  mayDeleteItems?: boolean;
  mayChangeAmountItems?: boolean;
  footer?: React.ReactNode;
  cssModificationClass?: string;
  size?: 'large';
}

interface IConnectedState {
  ui: Alicanto.Models.UI;
  items: Alicanto.Models.CatalogItem[];
}

type IProps = IOwnProps & IConnectedState;

interface IState {}

const mapStateToProps = (state: Store.IState) => ({
  ui: state.ui,
  items: state.cart.items,
});

const WrappedCartItems = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class CartItems extends React.Component<IProps, IState> {
    getDataSource = (): IRecord[] => {
      return this.props.items.map(item => {
        const uid = item.item;
        const count = _.get(item, 'count', 0);
        const schemaFields = item.schema_fields;
        const pictureUrl = _.get(schemaFields, 'picture_url.value', '');
        const name = _.get(schemaFields, 'name.value', '');
        const shortDescription = _.get(schemaFields, 'short_description.value', '');
        const currency = _.get(schemaFields, 'currency.value', '');
        const price = _.get(schemaFields, 'price.value', 0);
        return {
          pictureUrl,
          count,
          uid,
          cost: { price, currency },
          nameFields: { shortDescription, name },
          key: uid,
        };
      });
    };

    render() {
      const { mayChangeAmountItems, mayDeleteItems, cssModificationClass, size, items, footer } = this.props;
      const cartUI = this.props.ui.routes.cart;
      const columns = [
        {
          key: 'image',
          className: 'image-col',
          render: (text: string, record: IRecord) => {
            const { pictureUrl, nameFields: { name } } = record;
            return <CartProductImage pictureUrl={pictureUrl} imageAlt={name} />;
          },
        },
        {
          key: 'name',
          className: 'name-col',
          render: (text: string, record: IRecord) => {
            const { uid, nameFields: { name, shortDescription } } = record;
            return <Name uid={uid} name={name} shortDescription={shortDescription} />;
          },
        },
        {
          key: 'count',
          className: 'count-col',
          render: (text: string, record: IRecord) => {
            const { uid, count } = record;
            return <Amount mayChangeAmountItems={mayChangeAmountItems} uid={uid} count={count} />;
          },
        },
        {
          key: 'cost',
          className: 'cost-col',
          render: (text: string, record: IRecord) => {
            const { price, currency } = record.cost;
            return <Price price={price} currency={currency} />;
          },
        },
        {
          key: 'delete',
          className: 'delete-col',
          render: (text: string, record: IRecord) => {
            const { uid } = record;
            return <Delete mayDeleteItems={mayDeleteItems} uid={uid} />;
          },
        },
      ];
      return (
        <div
          className={cx('root', {
            [cssModificationClass]: !!cssModificationClass,
            large: size === 'large',
          })}
        >
          <style jsx>{`
            .root {
              width: 540px;
              max-width: 100%;
              font-size: 14px;
              font-weight: bold;
              padding: 15px 0;
            }
            .large {
              width: 100%;
              padding-bottom: 0;
            }
            .root :global(.list) {
              margin-bottom: 16px;
            }
            .root :global(.name-col) {
              width: 100%;
              max-width: 0;
            }
            .root :global(.image-col) {
              padding-left: 0;
            }
            .root :global(.delete-col) {
              padding-right: 0;
            }
            .root :global(.cost-col) {
              text-align: right;
              white-space: nowrap;
            }
            .large :global(.cost-col) {
              padding-left: 30px;
            }
            .root :global(.ant-row-flex:not(:last-of-type)) {
              margin-bottom: 9px;
              padding-bottom: 8px;
            }
            .root :global(.button-wrapper) {
              text-align: right;
            }
            .root :global(.button) {
              min-width: 192px;
              letter-spacing: 2px;
              text-transform: uppercase;
              font-weight: 500;
              font-size: 11px;
              line-height: 32px;
              text-decoration: none;
            }
            .root :global(.button + .button) {
              margin-left: 12px;
            }
            .has-border-bottom {
              border-bottom: 2px solid #e5ecf4;
            }
            .has-border-bottom :global(.ant-table-tbody > tr:not(:last-of-type) > td) {
              border-bottom: 1px solid #e5ecf4;
            }
            .root :global(.ant-table-tbody > tr > td) {
              padding-top: 9px;
              padding-bottom: 9px;
              border: 0;
            }
            .root :global(.ant-table-tbody > tr:hover > td) {
              background: transparent;
            }
          `}</style>
          {_.isEmpty(items) ? (
            <div className="app-text-empty">{cartUI.emptyStatus.i18n.defaultText}</div>
          ) : (
            <div className="inner">
              <div className="list">
                <Table columns={columns} dataSource={this.getDataSource()} pagination={false} showHeader={false} />
              </div>
              {footer && <div className="button-wrapper">{footer}</div>}
            </div>
          )}
        </div>
      );
    }
  },
);

export default WrappedCartItems;
