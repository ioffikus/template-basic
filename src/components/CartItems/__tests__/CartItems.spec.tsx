import React from 'react';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import uiStateMock from 'src/store/__mocks__/uiStore';
import cartStateMock from 'src/store/__mocks__/cartStore';
import { initialState as cartInitialState } from 'src/ducks/cart';
import CartItems from '../CartItems';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe(null, () => {
  it('render() with items', () => {
    const store = mockStore({ ui: uiStateMock, cart: cartStateMock });
    const wrapper = shallow(<CartItems />, { context: { store } });
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('render() without items', () => {
    const store = mockStore({ ui: uiStateMock, cart: cartInitialState });
    const wrapper = shallow(<CartItems />, { context: { store } });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
