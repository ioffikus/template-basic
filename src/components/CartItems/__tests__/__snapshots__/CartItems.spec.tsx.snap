// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() with items 1`] = `
<div
  className="jsx-615831497 root"
>
  <JSXStyle
    css=".root.jsx-615831497{width:540px;max-width:100%;font-size:14px;font-weight:bold;padding:15px 0;}.large.jsx-615831497{width:100%;padding-bottom:0;}.root.jsx-615831497 .list{margin-bottom:16px;}.root.jsx-615831497 .name-col{width:100%;max-width:0;}.root.jsx-615831497 .image-col{padding-left:0;}.root.jsx-615831497 .delete-col{padding-right:0;}.root.jsx-615831497 .cost-col{text-align:right;white-space:nowrap;}.large.jsx-615831497 .cost-col{padding-left:30px;}.root.jsx-615831497 .ant-row-flex:not(:last-of-type){margin-bottom:9px;padding-bottom:8px;}.root.jsx-615831497 .button-wrapper{text-align:right;}.root.jsx-615831497 .button{min-width:192px;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;text-transform:uppercase;font-weight:500;font-size:11px;line-height:32px;text-decoration:none;}.root.jsx-615831497 .button + .button{margin-left:12px;}.has-border-bottom.jsx-615831497{border-bottom:2px solid #e5ecf4;}.has-border-bottom.jsx-615831497 .ant-table-tbody > tr:not(:last-of-type) > td{border-bottom:1px solid #e5ecf4;}.root.jsx-615831497 .ant-table-tbody > tr > td{padding-top:9px;padding-bottom:9px;border:0;}.root.jsx-615831497 .ant-table-tbody > tr:hover > td{background:transparent;}"
    styleId="615831497"
  />
  <div
    className="jsx-615831497 inner"
  >
    <div
      className="jsx-615831497 list"
    >
      <Table
        bordered={false}
        className=""
        columns={
          Array [
            Object {
              "className": "image-col",
              "key": "image",
              "render": [Function],
            },
            Object {
              "className": "name-col",
              "key": "name",
              "render": [Function],
            },
            Object {
              "className": "count-col",
              "key": "count",
              "render": [Function],
            },
            Object {
              "className": "cost-col",
              "key": "cost",
              "render": [Function],
            },
            Object {
              "className": "delete-col",
              "key": "delete",
              "render": [Function],
            },
          ]
        }
        dataSource={
          Array [
            Object {
              "cost": Object {
                "currency": "€",
                "price": 0,
              },
              "count": 1,
              "key": "92c0d3cfa48940568421",
              "nameFields": Object {
                "name": "Starship Troopers. ePub. Free!",
                "shortDescription": "ePub version of Star",
              },
              "pictureUrl": "",
              "uid": "92c0d3cfa48940568421",
            },
            Object {
              "cost": Object {
                "currency": "€",
                "price": 14,
              },
              "count": 1,
              "key": "ba1d54813e9b410aa1ea",
              "nameFields": Object {
                "name": "Starship Troopers. ePub ru",
                "shortDescription": "ePub version of Star",
              },
              "pictureUrl": "",
              "uid": "ba1d54813e9b410aa1ea",
            },
            Object {
              "cost": Object {
                "currency": "€",
                "price": 11,
              },
              "count": 1,
              "key": "d6186dc8f72f4c9b932f",
              "nameFields": Object {
                "name": "Starship Troopers. PDF ru",
                "shortDescription": "PDF version of Stars",
              },
              "pictureUrl": "",
              "uid": "d6186dc8f72f4c9b932f",
            },
          ]
        }
        indentSize={20}
        loading={false}
        locale={Object {}}
        pagination={false}
        prefixCls="ant-table"
        rowKey="key"
        rowSelection={null}
        showHeader={false}
        size="large"
        useFixedHeader={false}
      />
    </div>
  </div>
</div>
`;

exports[`render() without items 1`] = `
<div
  className="jsx-615831497 root"
>
  <JSXStyle
    css=".root.jsx-615831497{width:540px;max-width:100%;font-size:14px;font-weight:bold;padding:15px 0;}.large.jsx-615831497{width:100%;padding-bottom:0;}.root.jsx-615831497 .list{margin-bottom:16px;}.root.jsx-615831497 .name-col{width:100%;max-width:0;}.root.jsx-615831497 .image-col{padding-left:0;}.root.jsx-615831497 .delete-col{padding-right:0;}.root.jsx-615831497 .cost-col{text-align:right;white-space:nowrap;}.large.jsx-615831497 .cost-col{padding-left:30px;}.root.jsx-615831497 .ant-row-flex:not(:last-of-type){margin-bottom:9px;padding-bottom:8px;}.root.jsx-615831497 .button-wrapper{text-align:right;}.root.jsx-615831497 .button{min-width:192px;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;text-transform:uppercase;font-weight:500;font-size:11px;line-height:32px;text-decoration:none;}.root.jsx-615831497 .button + .button{margin-left:12px;}.has-border-bottom.jsx-615831497{border-bottom:2px solid #e5ecf4;}.has-border-bottom.jsx-615831497 .ant-table-tbody > tr:not(:last-of-type) > td{border-bottom:1px solid #e5ecf4;}.root.jsx-615831497 .ant-table-tbody > tr > td{padding-top:9px;padding-bottom:9px;border:0;}.root.jsx-615831497 .ant-table-tbody > tr:hover > td{background:transparent;}"
    styleId="615831497"
  />
  <div
    className="jsx-615831497 app-text-empty"
  >
    Please add a product to card
  </div>
</div>
`;
